### PR TITLE
Rewrite SimSiam tutorial with low-level blocks

### LINKED ISSUE #554: Rewrite SimSiam Tutorial with Low-Level Building Blocks
Rewrite the SimSiam tutorial with low-level building blocks as we want to remove mid-level blocks (#493)
Specifically:

- [x] Add code for SimSiam model
- [x] Change loss to NegativeCosineSimilarity
- [x] Check that text is consistent with changes

### PR DESCRIPTION
**Pull request by guarin**
Closes #554 

- Remove lightly.modelsSimSiam dependency
- Add SimSiam module
- Remove SymNegCosineSimilarity dependency
- Add NegativeCosineSimilarity

[simsiam_esa.pdf](https://github.com/lightly-ai/lightly/files/7586401/simsiam_esa.pdf)

### CODE REVIEW

File: docs/source/tutorials_source/package/tutorial_simsiam_esa.py [hunk 1]
```diff
@@ -227,21 +238,20 @@
...
         # run the model on both transforms of the images
-        # the output of the simsiam model is a y containing the predictions
-        # and projections for each input x
-        y0, y1 = model(x0, x1)
+        # we get predictions and projections as output
+        p0, p1, z0, z1 = model(x0, x1)
 
-        # backpropagation
-        loss = criterion(y0, y1)
+        # apply the symmetric negatice cosine similarity
```
Comment (philippmwirth) [inline docs/source/tutorials_source/package/tutorial_simsiam_esa.py:244]: typo

**Review by philippmwirth — CHANGES_REQUESTED** [review]

File: docs/source/tutorials_source/package/tutorial_simsiam_esa.py (hunk 1 above)
Comment (guarin) [reply docs/source/tutorials_source/package/tutorial_simsiam_esa.py]: Thanks! Fixed :)

**Review by philippmwirth — APPROVED** [review]
well done
